python-package/pyproject.toml

```diff
@@ -57,21 +57,21 @@ changelog = "https://github.com/microsoft/LightGBM/releases"
 # start:build-system
 [build-system]
 
-requires = ["scikit-build-core>=0.9.3"]
+requires = ["scikit-build-core>=0.10.1"]
 build-backend = "scikit_build_core.build"
 
 # based on https://github.com/scikit-build/scikit-build-core#configuration
 [tool.scikit-build]
 
-cmake.version = ">=3.28"
+cmake.version = "CMakeLists.txt"
```

Comment:
This is described in the blog post linked in the PR... `scikit-build-core` now reads `cmake_minimum_required()` from `CMakeLists.txt` and uses it to figure this out. So no more need to keep those 2 versions in sync!

I'm not sure how this would work if `cmake_minimum_required()` varied based on options, as this project used to do prior to #6260 .... but I hope we'll not have to go back to that in the future and can just keep having a single minimum version.